python/thunder/rdds/series.py

```diff
@@ -56,6 +56,22 @@ def index(self):
         if self._index is None:
             self.populateParamsFromFirstRecord()
         return self._index
+        
+    @index.setter
+    def index(self, value):
+        try:
+            value[0]
+        except:
+            raise ValueError("Index must be a non-empty indexable object")
+        if self._index is None:
+            self.populateParametersFromFirstRecord()
+        try:
+            lenMatch = len(value) == len(self._index)
+        except:
+            raise ValueError("Index must be an object with a length")
+        if not lenMatch:
+            raise ValueError("New index length must match that of the current index")
```

Comment:
I might be wrong, but I think this won't work. In the even that the if-statement fails, it will raise "blah blah", which the except will then catch, and raise "another blah"...so the first exception will never be raised. At least that's what my initial testing is telling me.